docs/en/pact-reference.md

```diff
@@ -599,6 +612,218 @@ Examples of valid keyset JSON productions:
 
 ```
 
+### Namespaces {#namespaces}
+
+Namespaces are [defined](pact-functions.html#define-namespace) by specifying a namespace name and [associating](pact-functions.html#read-keyset)
+a keyset with the namespace. Namespace scope is entered by [declaring](pact-functions.html#namespace) the namespace environment. All definitions issued after the namespace scope is entered will be accessible by their fully qualified names. These names are of the form _namespace.module.definition_. This form can also be used to access code outside of the current namespace for the purpose of importing module code, or implementing modules:
+
+```lisp
+(implements my-namespace.my-interface)
+;; or
+(use my-namespace.my-module)
+```
+
+Code may be appended to the namespace by simply entering the re-entering the namespace and declaring new code definitions. All definitions _must_ occur within a namespace, as the global namespace (the empty namespace) is reserved for Kadena code.
+
+Examples of valid namespace definition and scoping:
+
+#### Example: Defining a namespace
+
+Defining a namespace requires a keyset, and a namespace name of type string:
+
+```lisp
```

Comment:
It's not the practice to give full-on repl examples. However I'd like to keep these around so maybe add to `pact-examples` repo? @LindaOrtega can give you details.

Reply:
```
pact> (define-namespace 'my-namespace (read-keyset 'my-keyset))
"[defined message]"
pact> (namespace 'my-namespace)
"Namespace set to my-namespace"
```

Reply:
Fixed. I'll come up with some examples for `pact-examples` that are less simplistic